inspect_buildpack_test.go

```diff
@@ -0,0 +1,490 @@
+										},
+										Optional: false,
+									},
+								},
+							},
+							{
+								Group: []dist.BuildpackRef{
+									{
+										BuildpackInfo: dist.BuildpackInfo{
+											ID:      "some/third-inner-buildpack",
+											Version: "3.0.0",
+										},
+										Optional: false,
+									},
+								},
+							},
+						},
+						LayerDiffID: "sha256:first-inner-buildpack-diff-id",
+						Homepage:    "first-inner-buildpack-homepage",
+					},
+				},
+				"some/second-inner-buildpack": {
+					"2.0.0": {
+						API: apiVersion,
+						Stacks: []dist.Stack{
+							{ID: "io.buildpacks.stacks.first-stack"},
+							{ID: "io.buildpacks.stacks.second-stack"},
+						},
+						LayerDiffID: "sha256:second-inner-buildpack-diff-id",
+						Homepage:    "second-inner-buildpack-homepage",
+					},
+				},
+				"some/third-inner-buildpack": {
+					"3.0.0": {
+						API: apiVersion,
+						Stacks: []dist.Stack{
+							{ID: "io.buildpacks.stacks.first-stack"},
+							{ID: "io.buildpacks.stacks.second-stack"},
+						},
+						LayerDiffID: "sha256:third-inner-buildpack-diff-id",
+						Homepage:    "third-inner-buildpack-homepage",
+					},
+				},
+				"some/top-buildpack": {
+					"0.0.1": {
+						API: apiVersion,
+						Order: dist.Order{
+							{
+								Group: []dist.BuildpackRef{
+									{
+										BuildpackInfo: dist.BuildpackInfo{
+											ID:      "some/first-inner-buildpack",
+											Version: "1.0.0",
+										},
+										Optional: false,
+									},
+									{
+										BuildpackInfo: dist.BuildpackInfo{
+											ID:      "some/second-inner-buildpack",
+											Version: "2.0.0",
+										},
+										Optional: false,
+									},
+								},
+							},
+							{
+								Group: []dist.BuildpackRef{
+									{
+										BuildpackInfo: dist.BuildpackInfo{
+											ID:      "some/first-inner-buildpack",
+											Version: "1.0.0",
+										},
+										Optional: false,
+									},
+								},
+							},
+						},
+						LayerDiffID: "sha256:top-buildpack-diff-id",
+						Homepage:    "top-buildpack-homepage",
+					},
+				},
+			},
+		}
+	})
+
+	it.After(func() {
+		mockController.Finish()
+		h.AssertNil(t, os.RemoveAll(tmpDir))
+	})
+
+	when("inspect-buildpack", func() {
+		when("inspecting a registry buildpack", func() {
+			var registryFixture string
+			it.Before(func() {
+				expectedInfo.Location = buildpack.RegistryLocator
+
+				registryFixture = h.CreateRegistryFixture(t, tmpDir, filepath.Join("testdata", "registry"))
+				packHome := filepath.Join(tmpDir, "packHome")
+				h.AssertNil(t, os.Setenv("PACK_HOME", packHome))
+				mockImageFetcher.EXPECT().Fetch(
+					gomock.Any(),
+					"example.com/some/package@sha256:8c27fe111c11b722081701dfed3bd55e039b9ce92865473cf4cdfa918071c566",
+					false,
+					config.PullNever).Return(buildpackImage, nil)
+			})
+
+			it.After(func() {
+				h.AssertNil(t, os.Unsetenv("PACK_HOME"))
+			})
+
+			it("succeeds", func() {
```

Comment:
nit: Could you also add a test for the case where it errors returning from the registry?